src/vs/workbench/contrib/chat/common/promptSyntax/service/promptsService.ts

```diff
@@ -79,14 +82,31 @@ export class PromptsService extends Disposable implements IPromptsService {
 	}
 
 	public async listPromptFiles(): Promise<readonly IPrompt[]> {
-		const promptFiles = await this.fileLocator.listFiles([]);
+		const globalLocations = [this.userDataService.currentProfile.promptsHome];
 
-		return promptFiles.map((uri) => {
+		const prompts = await Promise.all([
+			this.fileLocator.listFilesIn(globalLocations, [])
+				.then(withSource('global')),
+			this.fileLocator.listFiles([])
+				.then(withSource('local')),
+		]);
+
+		return prompts.flat();
+	}
+}
+
+/**
+ * Utility to add a provided prompt `source` to a list of prompt URIs.
+ */
+const withSource = (
```

Comment:
Is this same as the code below?:
```
const withSource = (source: IPrompt['source']) => (uris: readonly URI[]): readonly IPrompt[] =>
    uris.map(uri => ({ uri, source }));
```

Reply:
Yeah, but I prefer the arrow functions to be as unambiguous as possible 🤓 
I could make it a bit shorter in the last return without sacrifices though 🤔 e.g.,

```typescript
/**
 * Utility to add a provided prompt `source` to a list of prompt URIs.
 */
const withSource = (
	source: IPrompt['source'],
): (uris: readonly URI[]) => (readonly IPrompt[]) => {
	return (uris) => {
		return uris.map((uri) => {
			return { uri, source }; // here
		});
	};
};
```

Will update that in one of the upcoming PRs!